fault-remediation/pkg/reconciler/reconciler.go

```diff
@@ -502,3 +464,71 @@ func (r *Reconciler) parseHealthEvent(eventWithToken datastore.EventWithToken,
 
 	return result, nil
 }
+
+func (r *FaultRemediationReconciler) CloseAll(ctx context.Context) error {
+	if err := r.ds.Close(ctx); err != nil {
+		slog.Error("failed to close datastore", "error", err)
+		return err
+	}
+
+	if err := r.watcher.Close(ctx); err != nil {
+		slog.Error("failed to close watcher", "error", err)
+		return err
+	}
+	return nil
+}
+
+// SetupWithManager wires the reconciler + mongo stream
+func (r *FaultRemediationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ctx := context.Background()
+
+	r.watcher.Start(ctx)
+
+	reconciler := builder.TypedControllerManagedBy[*datastore.EventWithToken](mgr)
+	typedCh := AdaptEvents(ctx, r.watcher.Events())
+
+	//TODO: can we handle mark processsed ctrl runtime native?
+	//TODO: make this more generic/sharable for other controllers
+
+	src := source.TypedChannel[*datastore.EventWithToken, *datastore.EventWithToken](
+		typedCh,
+		handler.TypedFuncs[*datastore.EventWithToken, *datastore.EventWithToken]{
+			GenericFunc: func(
+				ctx context.Context,
+				e event.TypedGenericEvent[*datastore.EventWithToken],
+				q workqueue.TypedRateLimitingInterface[*datastore.EventWithToken],
+			) {
+				q.Add(e.Object)
+			},
+		},
+	)
+
+	return reconciler.
+		Named("fault-remediation-controller").
+		WatchesRawSource(
+			src,
+		).
+		Complete(r)
+}
+
+// TODO: make this part of the datastore interface?
```

Comment:
maybe not part of interface but at least should be reusable so each controller doesn't have to re-implement this. 

I'll leave a lot of stuff in this MR potentially less generic than it could be and then if we like this pattern and want to apply it to other controllers can focus and reusing code at that point